### PR TITLE
Add Solution To Project Euler Problem 50

### PR DESCRIPTION
**Pull request by abhishekjiitr**
Signed-off-by: Abhishek Jaisingh <abhi2254015@gmail.com>

### **Describe your change:**

* [x] Add an algorithm?
### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).

### CODE REVIEW

**Comment by abhishekjiitr** [comment]
@dhruvmanila can you review?

**Review by dhruvmanila — COMMENTED** [review]
### Announcement:

@abhishekjiitr

*This message is sent to all pull requests which are **OPEN, NOT MERGED, not containing 'wontfix' label and contains 'Project Euler' or 'Project' or 'Euler' (case doesn't matter) in their title**. If this message doesn't apply to your pull request, please ignore this.*

### Message:

This is to notify all the users submitting a pull request for the Project Euler solution that the [README.md](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) file has been updated with all the style requirements necessary for the directory and solution files. ***Please read through it and make all the necessary changes required to your submission.***

**Comment by abhishekjiitr** [comment]
@dhruvmanila  @Kush1101 required changes added.

**Review by 53jk1 — APPROVED** [review]

File: project_euler/problem_050/sol1.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            if p in prime_set:
+                curr_l = j - i + 1
+                if curr_l > max_l:
+                    max_prime = p
+                    max_l = curr_l
+    return max_prime
+
+
+# if __name__ == "__main__":
+#     print(solution())
```
Comment (poyea) [inline project_euler/problem_050/sol1.py:85]:
```suggestion

```

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

File: project_euler/problem_050/sol1.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            if p in prime_set:
+                curr_l = j - i + 1
+                if curr_l > max_l:
+                    max_prime = p
+                    max_l = curr_l
+    return max_prime
+
+
+if __name__ == "__main__":
+    print(f"solution = {solution()}")
```
Comment (dhruvmanila) [inline project_euler/problem_050/sol1.py:85]:
```suggestion
    print(f"{solution() = }")
```

File: project_euler/problem_050/sol1.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    >>> solution(1000)
+    953
+    >>> solution(1000000)
+    997651
+    """
+    prime_set, primes = get_primes(limit)
+    prefix = []
+
+    curr = 0
```
Comment (dhruvmanila) [inline project_euler/problem_050/sol1.py:56]: Descriptive variable names than `curr`, `p`, `l`, `max_l`...

File: project_euler/problem_050/sol1.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+def get_primes(limit: int) -> (set, list):
+    """
+    Returns the set & list(ordered) of primes up to the below limit
+    using Sieve of Eratosthenes
+
+    >> get_primes(10)
+    ({2, 3, 5, 7}, [2, 3, 5, 7])
+    """
+    SV_MAX = limit
```
Comment (dhruvmanila) [inline project_euler/problem_050/sol1.py:23]: You can directly use the `limit` parameter.

File: project_euler/problem_050/sol1.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+41 = 2 + 3 + 5 + 7 + 11 + 13
+This is the longest sum of consecutive primes that adds to a prime below one-hundred.
+
+The longest sum of consecutive primes below one-thousand that adds to a prime, contains 21 terms, and is equal to 953.
+
+Which prime, below one-million, can be written as the sum of the most consecutive primes?
+"""
+
+
+def get_primes(limit: int) -> (set, list):
```
Comment (dhruvmanila) [inline project_euler/problem_050/sol1.py:15]: Please elaborate on the type hints. Eg: `Set[type, type,...]`

File: project_euler/problem_050/sol1.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    """
+    Returns the set & list(ordered) of primes up to the below limit
+    using Sieve of Eratosthenes
+
+    >> get_primes(10)
+    ({2, 3, 5, 7}, [2, 3, 5, 7])
+    """
+    SV_MAX = limit
+    sv = [True for i in range(SV_MAX)]
```
Comment (dhruvmanila) [inline project_euler/problem_050/sol1.py:24]: Descriptive variable names.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: project_euler/problem_050/sol1.py (hunk 2 above)
Comment (abhishekjiitr) [reply project_euler/problem_050/sol1.py:85]: I am not sure what the change should be here. The code you suggest has invalid syntax:
```
❯❯ python3 sol1.py
  File "<fstring>", line 1
    (solution() = )
                ^
SyntaxError: invalid syntax
```

File: project_euler/problem_050/sol1.py (hunk 6 above)
Comment (abhishekjiitr) [reply project_euler/problem_050/sol1.py]: done

File: project_euler/problem_050/sol1.py (hunk 5 above)
Comment (abhishekjiitr) [reply project_euler/problem_050/sol1.py]: added further type hints for complex types

File: project_euler/problem_050/sol1.py (hunk 4 above)
Comment (abhishekjiitr) [reply project_euler/problem_050/sol1.py]: fixed

File: project_euler/problem_050/sol1.py (hunk 3 above)
Comment (abhishekjiitr) [reply project_euler/problem_050/sol1.py]: added more descriptive names for above

File: project_euler/problem_050/sol1.py (hunk 2 above)
Comment (cclauss) [reply project_euler/problem_050/sol1.py:86]: You are not running on Python >= 3.8.

Comment (abhishekjiitr) [reply project_euler/problem_050/sol1.py]: Thanks @cclauss , its done now

**Comment by abhishekjiitr** [comment]
@dhruvmanila can you review again?

File: project_euler/problem_050/sol1.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+The longest sum of consecutive primes below one-thousand that adds to a prime, contains 21 terms, and is equal to 953.
+
+Which prime, below one-million, can be written as the sum of the most consecutive primes?
+"""
+
+from typing import List, Set
+
+
+def get_primes(limit: int) -> (Set[int], List[int]):
```
Comment (dhruvmanila) [inline project_euler/problem_050/sol1.py:17]:
```suggestion
def get_primes(limit: int) -> Tuple[Set[int], List[int]]:
```
Add `from typing import Tuple`

**Comment by dhruvmanila** [comment]
`pre-commit` checks are failing and your solution is incorrect as per the `validate-solutions` check.

**Pull request closed without merging**